internal/lockservice/simpleLockService.go

```diff
@@ -111,64 +131,107 @@ func NewObjectDescriptor(ObjectID string) *ObjectDescriptor {
 }
 
 // NewSimpleLockService creates and returns a new lock service ready to use.
-func NewSimpleLockService(log zerolog.Logger) *SimpleLockService {
+func NewSimpleLockService(log zerolog.Logger, leaseDuration time.Duration) *SimpleLockService {
 	safeLockMap := &SafeLockMap{
-		LockMap: make(map[string]string),
+		LockMap: make(map[string]*LockMapEntry),
 	}
+	safeLockMap.LeaseDuration = leaseDuration
 	return &SimpleLockService{
 		log:     log,
 		lockMap: safeLockMap,
 	}
 }
+func fmtDuration(d time.Duration) string {
+	d = d.Round(time.Minute)
+	h := d / time.Hour
+	d -= h * time.Hour
+	ms := d / time.Microsecond
+	return fmt.Sprintf("%02d:%02d", h, ms)
+}
+
+// hasLeaseExpired returns true if the lease for a lock has expired and
+// returns false if the lease is still valid
+func hasLeaseExpired(timestamp time.Time, lease time.Duration) bool {
+	intDuration := int64(time.Now().Sub(timestamp))
+	intLease := int64(lease)
+	fmt.Printf("%d %d \n", intDuration, intLease)
+	if time.Now().Sub(timestamp) > lease {
+		return true
+	}
+	return false
+}
 
 // Acquire function lets a client acquire a lock on an object.
+// This lock is valid for a fixed duration that is set in the SafeLockMap.LeaseDuration
+// field. Beyond this duration, the lock has expired and the entity that owned the lock
+// for this period can no longer release it. The lock is open for acquisition after it
+// has expired.
 func (ls *SimpleLockService) Acquire(sd Descriptors) error {
 	ls.lockMap.Mutex.Lock()
-	if _, ok := ls.lockMap.LockMap[sd.ID()]; ok {
+
+	// If the lock is not present in the LockMap or
+	// the lock has expired, then allow the acquisition
+	if _, ok := ls.lockMap.LockMap[sd.ID()]; !ok || hasLeaseExpired(ls.lockMap.LockMap[sd.ID()].timestamp, ls.lockMap.LeaseDuration) {
```

Comment:
Can this be made less messy?
Say have a `timestamp := ls.lockMap.LockMap[sd.ID()].timestamp` and this use it while returning too?